firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteIndexManager.java

```diff
@@ -206,29 +248,27 @@ public void updateIndexEntries(Collection<Document> documents) {
 
   @Override
   public List<FieldIndex> getFieldIndexes(String collectionGroup) {
-    // TODO(indexing): Memoize the field index configuration
-    List<FieldIndex> matchingFieldIndexes = new ArrayList<>();
-    db.query(
-            "SELECT index_id, collection_group, index_proto, update_time_seconds, update_time_nanos "
-                + "FROM index_configuration "
-                + "WHERE active = 1 AND collection_group = ?")
-        .binding(collectionGroup)
-        .forEach(
-            row -> {
-              try {
-                matchingFieldIndexes.add(
-                    serializer.decodeFieldIndex(
-                        row.getString(1),
-                        row.getInt(0),
-                        Index.parseFrom(row.getBlob(2)),
-                        row.getInt(3),
-                        row.getInt(4)));
-              } catch (InvalidProtocolBufferException e) {
-                throw fail("Failed to decode index: " + e);
-              }
-            });
+    hardAssert(started, "IndexManager not started");
+    return memoizedIndexes.get(collectionGroup);
+  }
+
+  /** Stores the index in the memoized indexes table. */
+  private void memoizeIndex(FieldIndex fieldIndex) {
+    List<FieldIndex> existingIndexes = memoizedIndexes.get(fieldIndex.getCollectionGroup());
+    if (existingIndexes == null) {
+      existingIndexes = new ArrayList<>();
+      memoizedIndexes.put(fieldIndex.getCollectionGroup(), existingIndexes);
+    }
+
+    for (FieldIndex existingIndex : existingIndexes) {
```

Comment:
I was debating doing this already but thought it might be a bit more work since we have to extract the values in a lot of places. It made the tests a bit uglier, but the implementation mostly stayed the same. Fixed.